src/tiled/createobjecttool.h

```diff
@@ -66,6 +69,8 @@ class CreateObjectTool : public AbstractObjectTool
     ObjectGroupItem *mObjectGroupItem;
     MapObjectItem *mNewMapObjectItem;
     MapObjectItem *mOverlayPolygonItem;
+    bool mExtending;
+    bool mExtendingFirst;
```

Comment:
You can't in general extend an object so I'd prefer these variables to be in the `CreateMultipointObjectTool`. At least `mExtendingFirst` could easily be moved there since it isn't used in the base class. Regarding `mExtending`, are there any functions that couldn't be just overridden?